### ISSUE TITLE
Cannot read property 'APP_ENGINE' of undefined

### ISSUE THREAD

**Issue by sushshring** [googleapis/nodejs-logging-bunyan#277]
Creating the express middleware for logging causes the application to crash, as the variable that APP_ENGINE is being read from is undefined.

#### Environment details

  - OS: AppEngine Standard Env
  - Node.js version: 8.x.x
  - npm version: 6.4.1
  - `@google-cloud/logging-bunyan` version: 0.10.0

#### Steps to reproduce

  1. Create the `express.middleware()` from the logging library
  2. See crash due to value APP_ENGINE being read from some undefined object.

#### Stacktrace
```
TypeError: Cannot read property 'APP_ENGINE' of undefined
    at Object.<anonymous> (..../node_modules/@google-cloud/logging-bunyan/build/src/middleware/express.js:55:50)
    at Generator.next (<anonymous>)
    at fulfilled (..../node_modules/@google-cloud/logging-bunyan/build/src/middleware/express.js:4:58)
    at propagateAslWrapper (..../node_modules/async-listener/index.js:504:23)
    at ..../node_modules/async-listener/glue.js:188:31
    at ..../node_modules/async-listener/index.js:541:70
    at ..../node_modules/async-listener/glue.js:188:31
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:229:7)
    at process.fallback (..../node_modules/async-listener/index.js:565:15)
```
I tested that this error vanishes on 0.9.5. I'm not sure if there is some breaking change introduced in the new release which needs me to change how I'm setting up the middleware.

Making sure to follow these steps will guarantee the quickest resolution possible.

Thanks!

**Comment by ofrobots** [comment]
@sushshring thanks for the bug report. Can you check and report what version of the `google-auth-library` module you have in your application?

**Comment by ofrobots** [comment]
This is most likely related to https://github.com/googleapis/nodejs-logging-bunyan/pull/251

**Comment by sushshring** [comment]
It is `google-auth-library@3.1.0`

**Comment by ofrobots** [comment]
Hmm.. that version should work. Are you sure there are no other copies of it more deeply nested in your `node_modules`?

I do see a bug in this repo: we require `google-auth-library` but have no dependency on it. This means that we will get whatever auth library that happened to exist in the system even if it is lower than the minimum version we wanted.

**Comment by ofrobots** [comment]
Would it be possible for you to provide your `package.json` and `package-lock.json`?

**Comment by sushshring** [comment]
My `npm list` shows that there is an older version:
```
@google-cloud/debug-agent@2.6.0
│ └─┬ @google-cloud/common@0.17.0
│   └─┬ google-auto-auth@0.10.1
│     └── google-auth-library@1.6.1
```
I don't know how this affects it

**Comment by sushshring** [comment]
> Would it be possible for you to provide your `package.json` and `package-lock.json`?

They're big files. I'd be happy to, but it'd be messy to do here. Any other way to share?

**Comment by ofrobots** [comment]
Your `npm list` output answers my question. That's the version I think we end up picking up depending on how npm is hoisting things.

**Comment by ofrobots** [comment]
Fix in #278.
